### PR TITLE
#342: feat: add support for json.type command

### PR DESCRIPTION
**Pull request by jujiale**
Link to #342 

-  [JSON.TYPE] command return the value type that [JSON.GET] command returns.

 the json value type is similar to the redis command [JSON.TYPE] return.

- added unit-test and integration tests

### CODE REVIEW

File: core/eval.go [hunk 1]
```diff
@@ -287,6 +287,61 @@ func evalGETDEL(args []string) []byte {
...
+	// Default path is root if not specified
+	path := defaultRootPath
+	if len(args) > 1 {
+		path = args[1]
+	}
+	// Retrieve the object from the database
+	obj := Get(key)
+	if obj == nil {
+		return RespNIL
+	}
```
Comment (JyotinderSingh) [inline core/eval.go:310]: Please use `assertType` and `assertEncoding` methods to validate the retrieved object's type and encoding.
Return error if the object doesn't have JSON type and encoding.

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks a ton for adding support for the `JSON.TYPE` command @jujiale! The PR is very well written, left a few minor comments.

File: server/utils/jsontype.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+const (
+	ObjectType  string = "object"
+	ArrayType   string = "array"
+	StringType  string = "string"
+	IntegerType string = "integer"
+	NumberType  string = "number"
+	BooleanType string = "boolean"
+	NullType    string = "null"
+	UnknownType string = "unknown"
```
[the remark this answers is not in the document]
Comment (jujiale) [reply server/utils/jsontype.go]: done

File: server/utils/jsontype_test.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+			input:    struct{}{},
+			wantType: "unknown",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GetJSONFieldType(tt.input)
+			if result != tt.wantType {
+				t.Errorf("GetJsonFieldType(%q) = (%v), want (%v)", tt.input, result, tt.wantType)
```
[the remark this answers is not in the document]
Comment (jujiale) [reply server/utils/jsontype_test.go]: done

File: core/eval.go (hunk 1 above)
Comment (jujiale) [reply core/eval.go:310]: done

**Comment by jujiale** [comment]
@JyotinderSingh hello, I have optimized the code, please review. thanks a lot

File: core/commands.go [hunk 4]
```diff
@@ -94,6 +94,16 @@ var (
...
+	jsontypeCmdMeta = DiceCmdMeta{
+		Name: "JSON.TYPE",
+		Info: `JSON.TYPE key [path]
+		Returns string reply for each path, specified as the value's type.
+		Returns RespNIL If the key doesn't exist.
+		Error reply: If the number of arguments is incorrect.`,
+		Eval:     evalJSONTYPE,
+		Arity:    -1,
+		KeySpecs: KeySpecs{BeginIndex: 1},
+	}
```
Comment (JyotinderSingh) [inline core/commands.go:106]: Arity will be -2, right?

Also please make add this new object to the bottom of the list (we run an automated script to generate docs and it requires new objects to be added to the bottom of the list)

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]

File: core/commands.go (hunk 4 above)
Comment (jujiale) [reply core/commands.go:106]: yes. I misunderstand the arity is args number here. I will fix it immediately

File: core/commands.go [hunk 5]
```diff
@@ -600,6 +610,7 @@ func init() {
 	diceCmds["MSET"] = msetCmdMeta
 	diceCmds["JSON.SET"] = jsonsetCmdMeta
 	diceCmds["JSON.GET"] = jsongetCmdMeta
+	diceCmds["JSON.TYPE"] = jsontypeCmdMeta
```
Comment (jujiale) [inline core/commands.go:613]: (**Also please add this new object to the bottom of the list**) is it refer this?

File: core/commands.go (hunk 4 above)
Comment (jujiale) [reply core/commands.go]: done

File: core/commands.go (hunk 5 above)
Comment (JyotinderSingh) [reply core/commands.go:613]: That comment can be ignored for now. Thanks for making the other changes!

**Review by JyotinderSingh — APPROVED** [review]
Thanks for adding support for JSON.TYPE commans @jujiale! Approved.

**Pull request merged**